session/memory.go

```diff
@@ -185,16 +185,19 @@ func (this *MemSessionContainer) Session(sessionId string) (Session, bool) {
 		return nil, false
 	}
 	this.rwm.RLock()
-	defer this.rwm.RUnlock()
 	var ss, ok = this.sessions[sessionId]
 	if ok {
 		if ss.Dead() {
+			this.rwm.RUnlock()
```

Comment:
RUnlock should unlock after `var ss, ok = this.sessions[sessionId]`. Don't call RUnlock twice.